tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -393,6 +452,204 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+		categoryName := "category" + suffix
+		tagName := "tag" + suffix
+
+		catID, tagID := createCategoryNTag(ctx, categoryName, tagName)
+		defer func() {
+			deleteCategoryNTag(ctx, catID, tagID)
+		}()
+
+		attachTagToDS(ctx, tagID, sharedvmfsURL)
+		defer func() {
+			detachTagFromDS(ctx, tagID, sharedvmfsURL)
+		}()
+
+		attachTagToDS(ctx, tagID, sharedvmfs2URL)
+		defer func() {
+			detachTagFromDS(ctx, tagID, sharedvmfs2URL)
+		}()
+
+		allocationTypes = []string{
+			lztAllocType,
+		}
+
+		ginkgo.By("create SPBM policy with thin/ezt volume allocation")
+		ginkgo.By("create a storage class with a SPBM policy created from step 1")
+		ginkgo.By("create a PVC each using the storage policy created from step 2")
+		var storageclass *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var err error
+		var policyID *pbmtypes.PbmProfileId
+		for _, at := range allocationTypes {
+			policyID, policyName = createVmfsStoragePolicy(
+				ctx, pc, at, map[string]string{categoryName: tagName})
+
+			defer func() {
+				deleteStoragePolicy(ctx, pc, policyID)
+			}()
+
+			policyNames = append(policyNames, policyName)
+		}
+
+		if supervisorCluster {
+			assignPolicyToWcpNamespace(client, ctx, namespace, policyNames)
+		} else if guestCluster {
+			_, svNamespace := getSvcClientAndNamespace()
+			assignPolicyToWcpNamespace(client, ctx, svNamespace, policyNames)
+		}
+
+		for _, policyName := range policyNames {
+			if vanillaCluster {
+				ginkgo.By("CNS_TEST: Running for vanilla k8s setup")
+				scParameters[scParamStoragePolicyName] = policyName
+				storageclass, pvclaim, err = createPVCAndStorageClass(client,
+					namespace, nil, nil, "", nil, "", false, "")
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			} else if supervisorCluster {
+				ginkgo.By("CNS_TEST: Running for WCP setup")
+				profileID := e2eVSphere.GetSpbmPolicyID(policyName)
+				scParameters[scParamStoragePolicyID] = profileID
+				// create resource quota
+				createResourceQuota(client, namespace, rqLimit, policyName)
+				storageclass, pvclaim, err = createPVCAndStorageClass(client, namespace, nil,
+					scParameters, "", nil, "", false, "", policyName)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			} else {
+				ginkgo.By("CNS_TEST: Running for GC setup")
+				createResourceQuota(client, namespace, rqLimit, policyName)
+				scParameters[svStorageClassName] = policyName
+				storageclass, err = client.StorageV1().StorageClasses().Get(ctx, policyName, metav1.GetOptions{})
+				if !apierrors.IsNotFound(err) {
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				}
+				pvclaim, err = createPVC(client, namespace, nil, "", storageclass, "")
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+			pvcs = append(pvcs, pvclaim)
+			pvclaims2d = append(pvclaims2d, []*v1.PersistentVolumeClaim{pvclaim})
+			scs = append(scs, storageclass)
+		}
+
+		defer func() {
+			ginkgo.By("Delete the SCs created in step 2")
+			for _, sc := range scs {
```

Comment:
we should not delete SC for WCP and GC right

Reply:
Done